lib/parse.ts

```diff
@@ -35,35 +35,37 @@ type EnvSchemaParse<S extends BaseEnvSchema> = (
 ) => [EnvSchemaPartialValues<S>, EnvSchemaMaybeErrors<S>];
 
 export default <S extends BaseEnvSchema>(
-  schema: Readonly<S>,
-  properties: Readonly<EnvSchemaProperties<S>>,
-  customize: EnvSchemaCustomParsers<S> | undefined,
-): EnvSchemaParse<S> => (
-  container: Readonly<Record<string, string | undefined>>,
-): [EnvSchemaPartialValues<S>, EnvSchemaMaybeErrors<S>] =>
-  properties.reduce(
-    (
-      [values, initialErrors],
-      [key, propertySchema],
-    ): [EnvSchemaPartialValues<S>, EnvSchemaMaybeErrors<S>] => {
-      type K = typeof key;
-      const str = container[key];
-      let errors = initialErrors;
-      if (typeof str === 'string') {
-        const parser =
-          // we already checked for not undefined, but TS doesn't get it :-(
-          ((customize && customize[key]) as EnvSchemaParserFn<S, K>) ||
-          ((defaultParser as unknown) as EnvSchemaParserFn<S, K>);
-        try {
-          const value = parser(str, propertySchema, key, schema);
-          // eslint-disable-next-line no-param-reassign
-          values[key] = value;
-        } catch (e) {
-          dbg(`failed to parse "${key}": ${e}`, e);
-          errors = addErrors(errors, key, e);
+    schema: Readonly<S>,
+    properties: Readonly<EnvSchemaProperties<S>>,
+    customize: EnvSchemaCustomParsers<S> | undefined,
+  ): EnvSchemaParse<S> =>
+  (
+    container: Readonly<Record<string, string | undefined>>,
+  ): [EnvSchemaPartialValues<S>, EnvSchemaMaybeErrors<S>] =>
+    properties.reduce(
+      (
+        [values, initialErrors],
+        [key, propertySchema],
+      ): [EnvSchemaPartialValues<S>, EnvSchemaMaybeErrors<S>] => {
+        type K = typeof key;
+        const str = container[key];
+        let errors = initialErrors;
+        if (typeof str === 'string') {
+          const parser =
+            // we already checked for not undefined, but TS doesn't get it :-(
+            ((customize && customize[key]) as EnvSchemaParserFn<S, K>) ||
```

Comment:
I think we can remove this comment and assertion. After updating typescript, this shouldn't occur anymore.
```suggestion
            (customize && customize[key]) ||
```

Reply:
Nice!